src/blockchain/Blockchain.ts

```diff
@@ -27,19 +27,70 @@ export type Stats = {
   nodeVersion: string
 }
 
+export type txStatusPromiseResolver = (result: SubmittableResult) => boolean
+
 export interface IBlockchainApi {
   api: ApiPromise
   portablegabi: gabi.Blockchain
 
   getStats(): Promise<Stats>
   listenToBlocks(listener: (header: Header) => void): Promise<() => void>
-  submitTx(
+  signTx(
     identity: Identity,
     tx: SubmittableExtrinsic
+  ): Promise<SubmittableExtrinsic>
+  submitTx(
+    identity: Identity,
+    tx: SubmittableExtrinsic,
+    resolveOn?: txStatusPromiseResolver
   ): Promise<SubmittableResult>
   getNonce(accountAddress: string): Promise<Codec>
 }
 
+export const AWAIT_READY: txStatusPromiseResolver = (result) =>
+  result.status.isReady
+export const AWAIT_IN_BLOCK: txStatusPromiseResolver = (result) =>
+  result.isInBlock
+export const AWAIT_FINALIZED: txStatusPromiseResolver = (result) =>
+  result.isFinalized
+
+export async function submitTx(
+  tx: SubmittableExtrinsic,
+  resolveOn: txStatusPromiseResolver = AWAIT_FINALIZED
+): Promise<SubmittableResult> {
+  log.info(`Submitting ${tx.method}`)
+  let unsubscribe: () => void
+  return new Promise<SubmittableResult>((resolve, reject) => {
+    tx.send((result) => {
+      log.info(`Got tx status '${result.status.type}'`)
+      if (ErrorHandler.extrinsicFailed(result)) {
+        log.warn(`Extrinsic execution failed`)
+        log.debug(`Transaction detail: ${JSON.stringify(result, null, 2)}`)
+        const extrinsicError: ExtrinsicError =
+          ErrorHandler.getExtrinsicError(result) || ERROR_UNKNOWN
+        log.warn(`Extrinsic error ocurred: ${extrinsicError}`)
+        reject(extrinsicError)
+      } else if (result.isError) {
+        reject(
+          new Error(`Transaction failed with status '${result.status.type}'`)
+        )
+      } else if (resolveOn(result)) {
+        resolve(result)
+      }
+    })
+      .then((cb) => {
```

Comment:
can we change the name from `cb`? I know it is a callback, but it seems not verbose enough.